version/version_test.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package version
+
+import (
+	mf "github.com/jcrossley3/manifestival"
+	"knative.dev/eventing-operator/test/e2e"
+	"path/filepath"
+	"runtime"
+	"testing"
+)
+
+func TestManifestVersionSame(t *testing.T) {
+	clients := e2e.Setup(t)
+
+	_, b, _, _ := runtime.Caller(0)
+	m, err := mf.NewManifest(filepath.Join((filepath.Dir(b)+"/.."), "cmd/manager/kodata/knative-eventing/"), false, clients.Config)
+	if err != nil {
+		t.Fatal("Failed to load manifest", err)
+	}
+
+	// example: v0.10.1
+	expectedLabelValue := "v" + Version
+
+	resources := m.Resources
+
+	for _, resource := range resources {
+		v, found := resource.GetLabels()["eventing.knative.dev/release"]
+		if !found {
+			// label is missing for this resource. we cannot do the check.
+			continue
+		}
+		if v != expectedLabelValue {
+			t.Errorf("Version info in manifest and operator don't match. got: %v, want: %v. Resource GVK: %v, Resource name: %v", v, expectedLabelValue,
```

Comment:
If version info doesn't match, it shows this message multiple times.
Alternatively, we can switch to failing eagerly with `Fatalf`